drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/utils/ScenarioSimulationUtils.java

```diff
@@ -15,56 +15,31 @@
  */
 package org.drools.workbench.screens.scenariosimulation.client.utils;
 
+import java.util.Arrays;
 import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
 
-import com.ait.lienzo.client.core.event.NodeMouseClickEvent;
-import com.google.gwt.dom.client.NativeEvent;
 import org.drools.workbench.screens.scenariosimulation.client.factories.FactoryProvider;
+import org.drools.workbench.screens.scenariosimulation.client.interfaces.TetraFunction;
 import org.drools.workbench.screens.scenariosimulation.client.metadata.ScenarioHeaderMetaData;
 import org.drools.workbench.screens.scenariosimulation.client.renderers.ScenarioGridColumnRenderer;
-import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGrid;
 import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGridColumn;
 import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGridLayer;
 import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGridPanel;
-import org.uberfire.ext.wires.core.grids.client.model.GridCell;
-import org.uberfire.ext.wires.core.grids.client.model.GridData;
+import org.uberfire.ext.wires.core.grids.client.model.GridColumn;
 import org.uberfire.ext.wires.core.grids.client.widget.dom.single.impl.TextBoxSingletonDOMElementFactory;
 
 public class ScenarioSimulationUtils {
 
+    private static final TetraFunction<String, String, String, TextBoxSingletonDOMElementFactory, ScenarioHeaderMetaData> SCENARIOHEADERMETADATA_FUNCTION = ScenarioHeaderMetaData::new;
+    private static final TetraFunction<String, String, String, TextBoxSingletonDOMElementFactory, List<GridColumn.HeaderMetaData>> SCENARIOHEADERMETADATALIST_FUNCTION = (columnId, columnTitle, columnGroup, factory) -> Arrays.asList(SCENARIOHEADERMETADATA_FUNCTION.apply(columnId, columnGroup,"", factory), SCENARIOHEADERMETADATA_FUNCTION.apply(columnId, columnTitle, columnGroup, factory));
```

Comment:
Could you please check if it would be possible to format this line differently? It's very long. :-)

Reply:
About long lines, fixed with
[DROOLS-2860] Reformatting long lines

About editing issue, actually I did not change/modify nothing of the original code, and currently it is "buggy". Correct editing (both of header and row cells) should be the matter of a following task.

Reply:
@gitgabrio, alright, thank you!